### PR TITLE
[AI-3657] Add Metrics Origins metadata for python checks

### PR DESCRIPTION
**Pull request by dkirov-dd**
### What does this PR do?
Add Metrics Origins metadata for python checks

### Motivation
Metrics Origins for core and jmx checks is already implemented, there was a need to add python checks as well

### Possible Drawbacks / Trade-offs
This will cause all new python integrations to require creating new values for the `MetricSource` and `origin.proto` enums.

### Describe how to test/QA your changes
`OriginSubProduct` and `OriginProductDetail` can be viewed using [scott's proxy-dumper](https://github.com/DataDog/datadog-agent-benchmarks/tree/main/docker/proxy-dumper).
Run the proxy-dumper and change dd_url to http://localhost:8081/.
Run the agent.
See that the correct `OriginSubProduct` and `OriginProductDetail` values being populated for the checks

### CODE REVIEW

**Review by scottopell — COMMENTED** [review]
Looks good overall! My main feedback is that we need to cut down the un-needed 'MetricSource's here and ensure that only those mappable to an origin enum are present.

**Review by scottopell — COMMENTED** [review]
Looks good to me, I didn't double check that all new sources are python checks, but the mapping looks correct, and the usage of the check's name seems reasonable to me assuming integrations-core is comfortable with that identifier.

One question: If a check is configured to use say the http_check but it sets a new `name` in the config, does that break this mapping? Or does that `name` not affect the check id?

**Comment by dkirov-dd** [comment]
> One question: If a check is configured to use say the http_check but it sets a new `name` in the config, does that break this mapping? Or does that `name` not affect the check id?

As discussed on Slack, [this](https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/python/test_loader.go#L197) indicates that changing the name in the config would change the check instance and check id accordingly.
So running a custom check with a python check name would not be possible.

**Review by scottopell — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
This'll be useful to have!

File: pkg/serializer/internal/metrics/origin_mapping.go [hunk 1]
```diff
@@ -76,42 +218,320 @@ func metricSourceToOriginService(ms metrics.MetricSource) int32 {
 		return 9
 	case metrics.MetricSourceUnknown:
 		return 0
+	case metrics.MetricSourceActiveDirectory:
+		return 10
 	case metrics.MetricSourceActivemq:
```
Comment (chouetz) [inline pkg/serializer/internal/metrics/origin_mapping.go:223]: Is it expected not to have `11`?

File: pkg/metrics/metricsource.go [hunk 2]
```diff
@@ -60,6 +60,149 @@ const (
...
+	MetricSourceVault
+	MetricSourceVertica
+	MetricSourceVoltdb
+	MetricSourceVsphere
+	MetricSourceWin32EventLog
+	MetricSourceWindowsPerformanceCounters
+	MetricSourceWindowsService
+	MetricSourceWmiCheck
+	MetricSourceYarn
+	MetricSourceZk
```
Comment (FlorentClarret) [inline pkg/metrics/metricsource.go:205]: Related to our discussion: you're missing a couple of checks (tekton, strimzi, cloudera, ray, torchserve or temporal to name a few) I believe

Comment (dkirov-dd) [reply pkg/metrics/metricsource.go:205]: That's right! These are all checks that do not have a protobuf value in [origin.proto](https://github.com/DataDog/dd-source/blob/9ab55c73336e28013cd749401cdcdff0e37e43d2/domains/metrics/shared/libs/proto/origin/origin.proto#L493). I am in the process of including them in a separate [PR](https://github.com/DataDog/dd-source/pull/99385/files).

File: pkg/serializer/internal/metrics/origin_mapping.go (hunk 1 above)
Comment (dkirov-dd) [reply pkg/serializer/internal/metrics/origin_mapping.go:223]: Not expected, good catch!

**Comment by FlorentClarret** [comment]
@dkirov-dd You need a [changelog](https://datadoghq.dev/datadog-agent/guidelines/contributing/#reno) and a milestone to merge it

File: releasenotes/notes/python_metrics_origins-fc6e79b40c33d94d.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+other:
+  - Added metrics origins for python integrations.
```
Comment (hestonhoffman) [inline releasenotes/notes/python_metrics_origins-fc6e79b40c33d94d.yaml:10]:
```suggestion
  - Added metrics origins for Python integrations.
```

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**